resolwe/flow/models.py

```diff
@@ -271,24 +270,20 @@ class Meta(BaseModel.Meta):
 
     """
 
+    requirements = JSONField(default=dict)
 
-def render_template(template_string, context):
-    """Render template based on Dango template language."""
-    template_headers = [
-        '{% load resource_filters %}',
-        '{% load process_fields %}',
-        '{% load mathfilters %}',
-    ]
 
-    custom_template_tags = getattr(settings, 'RESOLWE_CUSTOM_TEMPLATE_TAGS', [])
-    if not isinstance(custom_template_tags, list):
-        raise KeyError("`RESOLWE_CUSTOM_TEMPLATE_TAGS` setting must be a list.")
+def render_template(process, template_string, context):
+    """Render template using the specified expression engine."""
+    from .managers import manager
```

Comment:
I do not know the mechanics exactly, but somehow Celery copies the dependent modules of the script that it runs on the worker node. Say you run execution engine on the worker (`celery_run.delay(self.executor, data_id, script, verbosity)
 `). The engine that runs on the worker node spawns new processes. This would trigger a run of the manager on the worker node. The manager then resolves dependencies and runs the spawned processes. Celery somehow knows it has to copy the `managers/celery.py` to the worker node so it can be imported and executed. However you import the function in a method (like here) to avoid circular dependencies instead of on the top, this somehow messes up Celery. 

@dblenkus should know more about this and if we have a test for this in resolwe. Here is a test for demultiplexing: https://github.com/genialis/resolwe-bio/blob/master/resolwe_bio/tests/processes/test_demultiplex.py.